Talos no longer builds on MacOs

## Bug Report

Talosctl does not build on MacOs with Docker desktop via the buildx builder created with the command in the Makefile

The change after which the build stops working with the following error is 921e10254d443c459a9775368ca080ecba273321 (via [siderolabs/talos#10170](https://github.com/siderolabs/talos/pull/10170))
Reversing this change fixes the issue
```diff
- TOOLS ?= ghcr.io/siderolabs/tools:v1.10.0-alpha.0-3-g0393558
+ TOOLS ?= ghcr.io/siderolabs/tools:v1.10.0-alpha.0-6-gbc30a2a
```

Works as expected with `tools:v1.10.0-alpha.0-3-g0393558`, does no longer work with `tools:v1.10.0-alpha.0-4-g01568a5` so it seems that the breaking change was made in https://github.com/siderolabs/tools/pull/417

I'd love to debug further but this is beyond my expertise.

### Logs

```sh
$ make talosctl-darwin-arm64

...
3.918 go: downloading github.com/prometheus/procfs v0.7.3
3.952 go: downloading google.golang.org/protobuf v1.34.2
3.991 go: downloading github.com/mattn/go-runewidth v0.0.16
4.100 go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
4.180 go: downloading github.com/rivo/uniseg v0.4.7
6.029 # github.com/golangci/golangci-lint/cmd/golangci-lint
6.029 /toolchain/go/pkg/tool/linux_arm64/link: running gcc failed: exit status 1
6.029 /toolchain/bin/gcc -Wl,-z,now -Wl,-z,nocopyreloc -fuse-ld=gold -o $WORK/b001/exe/a.out -rdynamic /tmp/go-link-1468027469/go.o /tmp/go-link-1468027469/000000.o /tmp/go-link-1468027469/000001.o /tmp/go-link-1468027469/000002.o /tmp/go-link-1468027469/000003.o /tmp/go-link-1468027469/000004.o /tmp/go-link-1468027469/000005.o /tmp/go-link-1468027469/000006.o /tmp/go-link-1468027469/000007.o /tmp/go-link-1468027469/000008.o /tmp/go-link-1468027469/000009.o /tmp/go-link-1468027469/000010.o /tmp/go-link-1468027469/000011.o /tmp/go-link-1468027469/000012.o /tmp/go-link-1468027469/000013.o /tmp/go-link-1468027469/000014.o /tmp/go-link-1468027469/000015.o /tmp/go-link-1468027469/000016.o /tmp/go-link-1468027469/000017.o /tmp/go-link-1468027469/000018.o /tmp/go-link-1468027469/000019.o /tmp/go-link-1468027469/000020.o /tmp/go-link-1468027469/000021.o /tmp/go-link-1468027469/000022.o /tmp/go-link-1468027469/000023.o /tmp/go-link-1468027469/000024.o -O2 -g -lresolv -O2 -g -ldl -O2 -g -lpthread -O2 -g
6.029 collect2: fatal error: cannot find 'ld'
6.029 compilation terminated.
6.029 
------
Dockerfile:213
--------------------
 212 |     ARG GOLANGCILINT_VERSION
 213 | >>> RUN --mount=type=cache,target=/.cache go install github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCILINT_VERSION} \
 214 | >>>      && mv /go/bin/golangci-lint /toolchain/go/bin/golangci-lint
 215 |     ARG GOIMPORTS_VERSION
--------------------
ERROR: failed to solve: process "/bin/sh -c go install github.com/golangci/golangci-lint/cmd/golangci-lint@${GOLANGCILINT_VERSION} \t&& mv /go/bin/golangci-lint /toolchain/go/bin/golangci-lint" did not complete successfully: exit code: 1

View build details: docker-desktop://dashboard/build/local/local0/s455qhtqfpzx0nlyt7vqakui7
make[2]: *** [Makefile:333: target-talosctl-darwin-arm64] Error 1
```

### Environment

- Platform: MacOs Sequoia 15.2
- Docker Desktop 4.37.2
- Go 1.23.5

related: https://github.com/siderolabs/talos/issues/10187

--- a/Dockerfile
+++ b/Dockerfile
@@ -206,6 +206,7 @@ FROM --platform=${BUILDPLATFORM} $TOOLS AS tools
 ENV PATH=/toolchain/bin:/toolchain/go/bin
 ENV LD_LIBRARY_PATH=/toolchain/lib
 ENV GOTOOLCHAIN=local
+ENV CGO_ENABLED=0
 RUN ["/toolchain/bin/mkdir", "/bin", "/tmp"]
 RUN ["/toolchain/bin/ln", "-svf", "/toolchain/bin/bash", "/bin/sh"]
 RUN ["/toolchain/bin/ln", "-svf", "/toolchain/etc/ssl", "/etc/ssl"]

